docs/reference/persistence-configuration.txt

```diff
@@ -177,22 +188,70 @@ failing subsequent updates:
   band.with(collection: "artists") do |band_object|
     band_object.save!
   end
-  
+
   # This will not save - updates the collection "bands" which does not have
   # the Scuba band
   band.update_attribute(likes: 1000)
-  
+
   # This will update the document.
   band.with(collection: "artists") do |band_object|
     band_object.update_attribute(likes: 1000)
   end
 
-As of Mongoid 6.0, the ``with`` method must always be called with a block,
-and the temporary persistence context exists only for the duration of the block.
-This is because a new client is created under the covers with the options
-passed to ``with``. To ensure that this client is closed and its associated
-resources are freed, the scope when this client could be used must be
-well-defined.
+The ``with!`` method was also introduced on document and criteria instances,
+which modifies and returns those instances with the new persistence context.
+Note that, since document instances cannot be ``dup``'ed, since doing so
+changes its _id, in order to change a document instance's persistence context
+we have to modify the document itself. Therefore, calling the ``with`` method
+on document instances without providing a block is unsupported, and ``with!``
+must be used.
+
+The ``with`` command can also be used without a block:
+
+.. code-block:: ruby
+
+  begin
+    # The #with method on Model classes returns a criteria.
+    criteria = Band.with(database: "music-non-stop")
+    # => #<Mongoid::Criteria
+          selector: {}
+          options:  {}
+          class:    Band
+          embedded: false>
+
+    criteria.first
+  ensure
+    criteria.clear_persistance_context!
+  end
+
+  begin
+    # Since a document instance cannot be duped (doing so changes its _id), in
+    # order to change its persistence context, the document itself has to be
+    # modified. Use the #with! method instead, which modifies and returns the
+    # document instance itself.
+    band = Band.new
+    band.with!(client: :tertiary)
+    band.save!
+  ensure
+    band.clear_persistance_context!
+  end
+
+  begin
+    # The #with method on criteria returns a new criteria.
+    criteria = Band.all.with(collection: "artists")
+    band = criteria.first
+  ensure
+    criteria.clear_persistance_context!
+  end
+
+As of Mongoid 8.1, the ``with`` method can be called with or without a block.
+If it is called with a block, the temporary persistence context will exist only
+for the duration of the block. If the ``with`` method is called without a block,
+the ``clear_persistance_context!`` method must be called on the criteria or
```

Comment:
Should be `clear_persistence_context` with "e"

Reply:
```suggestion
the ``clear_persistence_context!`` method must be called on the criteria or
```

Reply:
good catch.

Reply:
Think you still have the same mistake in several more places. Please scan the codebase for it.